### PR TITLE
promise: ShadowNode use seperate profile instead of polluting jerry

### PR DESCRIPTION
**Pull request by algebrait**
##### Checklist

- [x] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added

### CODE REVIEW

**Comment by algebrait** [comment]
This PR will pass promise cases in jerry-tests

File: CMakeLists.txt [hunk 1]
```diff
@@ -55,7 +55,7 @@ if(NOT DEFINED ENABLE_LTO)
 endif()
 
 if(NOT DEFINED DISABLE_ES2015)
-  set(FEATURE_PROFILE "es2015-subset")
+  set(FEATURE_PROFILE "ShadowNode")
 endif()
```
Comment (yorkie) [inline CMakeLists.txt]: I think we can removed `DISABLE_ES2015` and use the `shadow-node` directly or by default.

File: deps/jerry/jerry-core/profiles/ShadowNode.profile [hunk 2]
```diff
@@ -0,0 +1,2 @@
+CONFIG_DISABLE_ES2015_PROMISE_BUILTIN
```
Comment (yorkie) [inline deps/jerry/jerry-core/profiles/ShadowNode.profile]: To keep the consistent with other profiles, suggestion rename: `shadow-node.profile`.

BTW, could you please also move the this profile to ShadowNode root directory, JerryScript supports the absolute path at [deps/jerry/jerry-core/CMakeLists.txt#L236-L247](https://github.com/Rokid/ShadowNode/blob/master/deps/jerry/jerry-core/CMakeLists.txt#L236-L247).

File: CMakeLists.txt [hunk 3]
```diff
@@ -54,9 +54,7 @@ if(NOT DEFINED ENABLE_LTO)
   set(ENABLE_LTO OFF)
 endif()
 
-if(NOT DEFINED DISABLE_ES2015)
-  set(FEATURE_PROFILE "es2015-subset")
-endif()
```
Comment (yorkie) [inline CMakeLists.txt]: The above lines should not be removed, I think adding the default FEATURE_PROFILE for ShadowNode would be better :)

File: profiles/jerry.profile [hunk 4]
```diff
@@ -0,0 +1,2 @@
+CONFIG_DISABLE_ES2015_PROMISE_BUILTIN
```
Comment (yorkie) [inline profiles/jerry.profile]: How about putting the profile on the root directory? Actually we won't own other `*.profile` files anyway.

Comment (algebrait) [reply profiles/jerry.profile]: Will ShadowNode use the same profile file on every target later? If not, I would prefer put it on profilers directory.

Comment (yorkie) [reply profiles/jerry.profile]: So how about profilers/default.profile?

Comment (algebrait) [reply profiles/jerry.profile]: Directory profiles already has default.profile and minimal.profile. They are for ShadowNode modules. Perhaps default.jerry.profile is better to show it is profile for jerry.

Comment (yorkie) [reply profiles/jerry.profile]: Why not `./profiles/jerry/default.profile`, an implicit separator would make confusion a bit to me :(

Comment (algebrait) [reply profiles/jerry.profile]: Well done, that's it!

**Comment by algebrait** [comment]
There is still a jerry unit test failed - test-snapshot.c.
Because #390 will change snapshot format, this PR will not give a temp patch.

**Pull request merged**